src/gpu/intel/jit/v2/conv/kernel_desc.cpp

```diff
@@ -301,11 +301,26 @@ bool fit_tag(tensor_kind_t abc, const kernel_desc_t &kernel_desc,
     return true;
 }
 
-bool fit_impl(const kernel_desc_t &desc, const problem_t &prb, bool exact) {
+bool is_compatible(const hw_desc_t &hw_desc, const hw_t &hw, bool exact) {
+    if (!exact && hw != hw_desc.hw) {
+        switch (hw_desc.hw) {
+            case ngen::HW::XeHPC:
+                return utils::one_of(
+                        hw.to_ngen(), ngen::HW::Xe2, ngen::HW::Xe3);
+            default: break;
+        }
+    }
+    return hw_desc.hw == hw.to_ngen();
+}
+
+bool is_compatible(
+        const kernel_desc_t &desc, const problem_t &prb, bool exact) {
+    ir_check(is_compatible(desc.hw_desc, prb.hw(), exact))
+            << "HW does not match";
     ir_check(prb.prop() == desc.prop) << "Propagation kind does not match";
-    ir_check(fit_tag(tensor_kind_t::a, desc, prb, exact));
-    ir_check(fit_tag(tensor_kind_t::b, desc, prb, exact));
-    ir_check(fit_tag(tensor_kind_t::c, desc, prb, exact));
+    ir_check(is_compatible(tensor_kind_t::a, desc, prb, exact));
+    ir_check(is_compatible(tensor_kind_t::b, desc, prb, exact));
+    ir_check(is_compatible(tensor_kind_t::c, desc, prb, exact));
     ir_check(prb.is_depthwise() == desc.is_dw)
             << "Mixing depthwise/non-depthwise descriptor and problem";
```

Comment:
Random spot: 2d messages have different alignment requirements on Xe2, but I don't see any check for it. Where is that check happening?

Reply:
Good point, thanks. I'll open another PR to refactor block 2D requirements to remove them from the kernel descriptor. This way we can tailor a kernel descriptor to the target GPU platform on the fly.

Reply:
Addressed in https://github.com/oneapi-src/oneDNN/pull/2545